core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -2193,7 +2195,11 @@ public InstallationJob(Plugin plugin, UpdateSite site, Authentication auth, bool
 
         @Override
         protected URL getURL() throws MalformedURLException {
-            return new URL(plugin.url);
+            try {
+                return new URI(plugin.url).toURL();
+            } catch (URISyntaxException e) {
+                throw new RuntimeException(e);
```

Comment:
callers of this already had to cope and adjust to the URL being invalid - they can not longer do this and their code will be broken as this will thow a RuntimeException and no longer MalformedURLException.

rather than throwing the RuntimeExeption throw a `MalformedURLException` with the appropriate message initialized from this exception

Reply:
Changed to
```
            try {
                return new URI(plugin.url).toURL();
            } catch (URISyntaxException e) {
                throw new MalformedURLException("Not a valid URL");
            }
```